### PR TITLE
Update docs around the Semgrep Changelog Process

### PR DESCRIPTION
**Pull request by spencerdrak**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs

### CODE REVIEW

**Review by brendongo — APPROVED** [review]

File: docs/contributing/contributing-code.md [hunk 1]
```diff
@@ -149,6 +149,16 @@ Once `pre-commit` is working you may commit code and create pull requests as
...
 [CI to be passing](https://github.com/returntocorp/semgrep/actions).
 
+In addition to the above, and depending on the nature of the change, you may want to add an entry to the changelog. A tool called `towncrier` is used for changelog management. There is a directory at the root of the repository, `changelog.d`, which contains individual changelog entries, each in their own file. An example of such a file is contained in the repository, but the format is described below as well.
+
+A changelog entry file named `cli-123.fixed` would indicate that the branch contains changes that addressed a CLI ticket number 123, and that the changes fixed existing functionality. A changlog entry file named `pa-321.added` would indicate that new functionality was added in support of PA ticket number 321. If you are an open-source contributor or otherwise don't have a ticket number to reference, you can also substitute the first part of the changelog entry filename with another string, such as a GitHub issue number or a semantic name, such as `gh-1234.added` or `logoutput.fixed`. 
+
+Acceptable suffixes include:
+- `added` - for new features or other previously non-existing functionality
```
Comment (adamkvitek) [inline docs/contributing/contributing-code.md:157]:
```suggestion
- `added` - New features or other previously non-existing functionality.
```

File: docs/contributing/contributing-code.md [hunk 2]
```diff
@@ -149,6 +149,16 @@ Once `pre-commit` is working you may commit code and create pull requests as
...
+- `changed` - for items that have changed the way semgrep functions
```
Comment (adamkvitek) [inline docs/contributing/contributing-code.md:158]:
```suggestion
- `changed` - Items that have changed the way Semgrep functions.
```

File: docs/contributing/contributing-code.md [hunk 3]
```diff
@@ -149,6 +149,16 @@ Once `pre-commit` is working you may commit code and create pull requests as
 you would expect. Pull requests require approval of at least one maintainer and
 [CI to be passing](https://github.com/returntocorp/semgrep/actions).
 
+In addition to the above, and depending on the nature of the change, you may want to add an entry to the changelog. A tool called `towncrier` is used for changelog management. There is a directory at the root of the repository, `changelog.d`, which contains individual changelog entries, each in their own file. An example of such a file is contained in the repository, but the format is described below as well.
+
+A changelog entry file named `cli-123.fixed` would indicate that the branch contains changes that addressed a CLI ticket number 123, and that the changes fixed existing functionality. A changlog entry file named `pa-321.added` would indicate that new functionality was added in support of PA ticket number 321. If you are an open-source contributor or otherwise don't have a ticket number to reference, you can also substitute the first part of the changelog entry filename with another string, such as a GitHub issue number or a semantic name, such as `gh-1234.added` or `logoutput.fixed`. 
+
+Acceptable suffixes include:
```
Comment (adamkvitek) [inline docs/contributing/contributing-code.md:156]:
```suggestion
Acceptable changelog file suffixes include:
```

File: docs/contributing/contributing-code.md [hunk 4]
```diff
@@ -149,6 +149,16 @@ Once `pre-commit` is working you may commit code and create pull requests as
...
+- `added` - for new features or other previously non-existing functionality
+- `changed` - for items that have changed the way semgrep functions
+- `fixed` - for bugfixes or other improvements
```
Comment (adamkvitek) [inline docs/contributing/contributing-code.md:159]:
```suggestion
- `fixed` - Bug fixes or other improvements.
```

File: docs/contributing/contributing-code.md [hunk 5]
```diff
@@ -149,6 +149,16 @@ Once `pre-commit` is working you may commit code and create pull requests as
...
+- `infra` - for workflow improvements or other non-code updates
```
Comment (adamkvitek) [inline docs/contributing/contributing-code.md:160]:
```suggestion
- `infra` - Workflow improvements or other non-code updates.
```

**Review by adamkvitek — COMMENTED** [review]
@spencerdrak thank you for this update! Made a few minor suggestions to make this text flow a little better. One major suggestion is for when to actually add a changelog entry. Thank you!

**Review by s-santillan — APPROVED** [review]
lgtm

**Review by adamkvitek — COMMENTED** [review]
Sorry @spencerdrak, I made this last comment review on vacation.

**Pull request merged**